packages/glsp-playwright-core/src/glsp/graph/decorators/decorator.ts

```diff
@@ -0,0 +1,41 @@
+import { PEdge, PEdgeConstructor, PModelElement, PModelElementConstructor, PNode, PNodeConstructor } from '~/glsp';
+import type { ConstructorT } from '~/types';
+import type { EdgeMetadata } from './edge.decorator';
+import type { ModelElementMetadata } from './html.decorator';
+import type { NodeMetadata } from './node.decorator';
+
+export const metadataKey = Symbol.for('metadata');
+
+export namespace PMetadata {
```

Comment:
This also needs way more documentation. Seems like a core concept that is used heavily.